static/skin/index.js

```diff
@@ -0,0 +1,31 @@
+const root = $(`link[type='root']`).attr('href');
+
+function getInnerHtml(node, query) {
+    return node.querySelector(query).innerHTML;
+}
+
+async function loadAndDisplayBooks() {
+  fetch(`${root}/catalog/search`).then(async (resp) => {
+    const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
+    const books = data.querySelectorAll('entry');
+    let bookHtml = '';
+    books.forEach((book) => {
+        const link = book.querySelector('link').getAttribute('href');
+        const title =  getInnerHtml(book, 'title');
+        const description = getInnerHtml(book, 'summary');
+
+        bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
+            <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
+            <div class='book__title' title='${title}'>${title}</div>
+            <div class='book__description' title='${description}'>${description}</div>
+            <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
+            </div>
+        </div></a>`;
```

Comment:
1. Please extract this into a function `generateBookHtml(book)` that produces HTML for a single book.
2. I would prefer that all `getInnerHtml()` calls are made outside the template literal. Expressions like `data-id='${getInnerHtml(book, 'id')}'` are a little difficult to parse visually because of nested single-quote pairs.

Reply:
> All the new global variables and functions in index.js increase the risk of interfering with variables/functions defined in other JS scripts (including those from the ZIM archives).

Thank you @veloman-yunkan. I don't want to interfer much in the technical review, but considering that this welcome page has only our code (and not code coming from any ZIM file). There is no risk I can see of interference with other content JS.

Reply:
@kelson42 he might  be talking about `block-external.js` other than that I agree it can be easily maintained other than that there are no other js files which will interfere not even `taskbar.js` as I have removed them for filtering part.